agent/site.py

```diff
@@ -459,11 +459,17 @@ def tablewise_backup(self):
         return data
 
     @step("Migrate Site")
-    def migrate(self, skip_failing_patches=False):
+    def migrate(self, skip_search_index=False, skip_failing_patches=False):
+        cmd = "migrate"
+        if skip_search_index:
+            cmd += " --skip-search-index"
         if skip_failing_patches:
-            return self.bench_execute("migrate --skip-failing")
-        else:
-            return self.bench_execute("migrate")
+            cmd += " --skip-failing"
+        return self.bench_execute(cmd)
+
+    @step("Build Search Index")
+    def build_search_index(self):
+        return self.bench_execute("build-search-index")
```

Comment:
This step takes a lot of time hence it's moved to background. 


The proper fix would be to not start background jobs until maintenance mode is disabled.  That way even if update fails or site is moved around the background job won't start.